aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1081 @@
+    const EPAYLOAD_NOT_STORED: u64 = 8;
+    /// Provided target function does not match the hash stored in the on-chain transaction.
+    const ETARGET_FUNCTION_DOES_NOT_MATCH_HASH: u64 = 9;
+    /// Provided arguments do not match the hash stored in the on-chain transaction.
+    const EARGUMENTS_DOES_NOT_MATCH_HASH: u64 = 10;
+    /// Transaction has not received enough approvals to be executed.
+    const ENOT_ENOUGH_APPROVALS: u64 = 11;
+    /// Transaction has not received enough rejections to be removed.
+    const ENOT_ENOUGH_REJECTIONS: u64 = 12;
+    /// Number of signatures required must be more than zero and at most the total number of owners.
+    const EINVALID_SIGNATURES_REQUIRED: u64 = 13;
+    /// Function args hash must be exactly 32 bytes (sha3-256).
+    const EINVALID_FUNCTION_ARGS_HASH: u64 = 14;
+
+    /// Represents a multisig account's configurations and transactions.
+    /// This will be stored in the multisig account (created as a resource account separate from any owner accounts).
+    struct MultisigAccount has key {
+        // The list of all owner addresses.
+        owners: vector<address>,
+        // The number of signatures required to pass a transaction (k in k-of-n).
+        signatures_required: u64,
+        // Map from transaction id (incrementing id) to transactions to execute for this multisig account.
+        // Already executed transactions are deleted to save on storage but can always be accessed via events.
+        transactions: Table<u64, MultisigTransaction>,
+        // Last executed or rejected transaction id. Used to enforce in-order executions of proposals.
+        last_transaction_id: u64,
+        // The transaction id to assign to the next transaction.
+        next_transaction_id: u64,
+        // The signer capability controlling the multisig (resource) account. This can be exchanged for the signer.
+        // Currently not used as the MultisigTransaction can validate and create a signer directly in the VM but
+        // this can be useful to have for on-chain composability in the future.
+        signer_cap: SignerCapability,
+
+        // Events.
+        add_owners_events: EventHandle<AddOwnersEvent>,
+        remove_owners_events: EventHandle<RemoveOwnersEvent>,
+        update_signature_required_events: EventHandle<UpdateSignaturesRequiredEvent>,
+        create_transaction_events: EventHandle<CreateTransactionEvent>,
+        approve_transaction_events: EventHandle<ApproveTransactionEvent>,
+        reject_transaction_events: EventHandle<RejectTransactionEvent>,
+        execute_transaction_events: EventHandle<ExecuteTransactionEvent>,
+        remove_transaction_events: EventHandle<RemoveTransactionEvent>,
+    }
+
+    /// A transaction to be executed in a multisig account.
+    /// This must contain either the full transaction payload or its hash (stored as bytes).
+    struct MultisigTransaction has copy, drop, store {
```

Comment:
how about add a expiration time?

Reply:
Is there a strong use case for an expiration time?